vulkan_generator/struct_factories/generator.py

```diff
@@ -0,0 +1,451 @@
+
+import abc
+
+from pathlib import Path
+from typing import List
+from typing import Dict
+
+from textwrap import dedent
+from vulkan_generator.vulkan_parser import types
+from vulkan_generator.codegen_utils import codegen
+
+
+def struct_factory_name(struct: str) -> str:
+    return struct + "Factory"
+
+
+def set_member_name(member: str) -> str:
+    return "set" + member[0:1].upper() + member[1:]
+
+
+def set_member_arg_type(member: types.VulkanStructMember, vulkan_metadata: types.VulkanMetadata) -> str:
+    remapped_type = remap_member_type(member.variable_type, member.variable_name, vulkan_metadata)
+    if "*" in remapped_type:
+        return remapped_type
+    else:
+        return "const " + remapped_type + "&"
+
+
+def get_member_name(member: str) -> str:
+    return "get" + member[0:1].upper() + member[1:]
+
+
+def get_member_return_type(member: types.VulkanStructMember, vulkan_metadata: types.VulkanMetadata) -> str:
+    remapped_type = remap_member_type(member.variable_type, member.variable_name, vulkan_metadata)
+    if "*" in remapped_type:
+        return remapped_type
+    else:
+        return "const " + remapped_type + "&"
+
+
+def tokenize_typename(member_type: str) -> List[str]:
+
+    tokens: List[str] = []
+    current_token = ""
+
+    for char in member_type:
+        if char.isspace():
+            if len(current_token) > 0:
+                tokens += [current_token]
+            current_token = ""
+        elif char == "*":
+            if len(current_token) > 0:
+                tokens += [current_token]
+            tokens += ["*"]
+            current_token = ""
+        else:
+            current_token += char
+
+    if len(current_token) > 0:
+        tokens += [current_token]
+
+    return tokens
+
+
+def sort_structs_dep_order_impl(struct :str,
+                     processed_structs : Dict[str, bool],
+                     sorted_structs : List[str],
+                     vulkan_metadata: types.VulkanMetadata) :
+
+    if struct in processed_structs:
+        return
+
+    for member in vulkan_metadata.types.structs[struct].members:
+        tokens = tokenize_typename(vulkan_metadata.types.structs[struct].members[member].variable_type)
+        for token in tokens:
+            if token in vulkan_metadata.types.structs and token != struct:
+                sort_structs_dep_order_impl(token, processed_structs, sorted_structs, vulkan_metadata)
+
+    processed_structs[struct] = True
+    sorted_structs += [struct]
+
+
+def sort_structs_dep_order(vulkan_metadata: types.VulkanMetadata) -> List[str] :
+    processed_structs : Dict[str, bool] = {}
+    sorted_structs : List[str] = []
+
+    for struct in vulkan_metadata.types.structs:
+        sort_structs_dep_order_impl(struct, processed_structs, sorted_structs, vulkan_metadata)
+
+    return sorted_structs
+
+
+def remap_member_type(member_type: str, member_name: str, vulkan_metadata: types.VulkanMetadata) -> str:
```

Comment:
I think that this function strips the typename from the typename. It will be unnecessary after merging my PR. After the postprocess you can access both the plain typename and the full typename in a struct.

e.g.
```
full_typename: "const VKSomethingCreateInfo*"
type.typename "VkSomethingCreateInfo"
```

So we should merge that PR first so that you don't invest much time on fixing these kind of issues that are already fixed in postprocess.